### PR TITLE
Minimal set of changes to get python 3 working

### PR DESCRIPTION
**Pull request by andrewmchen**
Made some tweaks to the code to get Databricks CLI working on Python 3. Tests passed but two commands still failed because of insufficient test coverage.
  - `databricks workspace import_dir`
  - `databricks fs cp local remote`
I applied fixes for both of these in this PR. I didn't add regression tests for these two cases yet because the error is in the SDK layer which we currently mock out of all unit tests...

Lots of ideas stolen from https://github.com/databricks/databricks-cli/pull/58/files. Thanks @andreabedini!

### CODE REVIEW

File: databricks_cli/dbfs/api.py [hunk 1]
```diff
@@ -99,7 +99,8 @@ def put_file(self, src_path, dbfs_path, overwrite):
                 contents = local_file.read(BUFFER_SIZE_BYTES)
                 if len(contents) == 0:
                     break
-                self.client.add_block(handle, b64encode(contents))
+                # add_block should not take a bytes object.
+                self.client.add_block(handle, b64encode(contents).decode())
```
Comment (mengxr) [inline databricks_cli/dbfs/api.py]: Is it just for py3 or a bug fix? If this is a correctness issue independent of py3 compatibility, we should fix it separately and make a bug fix release.

File: databricks_cli/sdk/api_client.py [hunk 2]
```diff
@@ -62,16 +62,8 @@ class ApiClient(object):
...
-        if configUrl:
-            self.url = configUrl
-            params = self.performQuery("/", headers = {})[1]
-            params = credential.json
-            user = str(params["user"])
-            password = str(params["password"])
-            host = str(params["apiUrl"].split("/api")[0])
-
+    def __init__(self, user = None, password = None, host = None, token = None,
+                 apiVersion = version.API_VERSION, default_headers = {}, verify = True):
```
Comment (mengxr) [inline databricks_cli/sdk/api_client.py]: Not part of this PR but `{}` shouldn't be the default value because it is immutable.

File: databricks_cli/sdk/api_client.py [hunk 3]
```diff
@@ -62,16 +62,8 @@ class ApiClient(object):
     A partial Python implementation of dbc rest api
     to be used by different versions of the client.
     """
-    def __init__(self, user = None, password = None, host = None, token = None, configUrl = None,
-            apiVersion = version.API_VERSION, default_headers = {}, verify = True):
-        if configUrl:
```
Comment (mengxr) [inline databricks_cli/sdk/api_client.py]: Why we no longer need this block?

File: tests/workspace/test_cli.py [hunk 4]
```diff
@@ -47,7 +46,7 @@ def test_export_workspace_cli(workspace_api_mock, tmpdir):
     workspace_api_mock.get_status.return_value = \
         WorkspaceFileInfo('/notebook-name', NOTEBOOK, WorkspaceLanguage.SCALA)
     runner = CliRunner()
-    runner.invoke(cli.export_workspace_cli, ['--format', 'SOURCE', '/notebook-name', path])
+    print(runner.invoke(cli.export_workspace_cli, ['--format', 'SOURCE', '/notebook-name', path]).output)
```
Comment (mengxr) [inline tests/workspace/test_cli.py]: What does `print` do here?

File: databricks_cli/dbfs/api.py (hunk 1 above)
Comment (andrewmchen) [reply databricks_cli/dbfs/api.py]: It's just for py3.

File: databricks_cli/sdk/api_client.py (hunk 3 above)
Comment (andrewmchen) [reply databricks_cli/sdk/api_client.py]: Turns out we just never used this piece of code. It's even a bit buggy. (credentials isn't defined).

File: tests/workspace/test_cli.py (hunk 4 above)
Comment (andrewmchen) [reply tests/workspace/test_cli.py]: ah this was for debugging. i need to revert it now.
